TemplePlus/temple_enums.cpp

```diff
@@ -455,8 +455,13 @@ static const char *ObjTypeNames[] = {
 	"obj_t_bag" // 16
 };
 
+static const char* unknownObject = "obj_t_unknown";
+
 const char* GetObjectTypeName(ObjectType type) {
-	return ObjTypeNames[type];
+	if (type < std::size(ObjTypeNames)) {
```

Comment:
Doesn't this just hide other bugs?

Reply:
Thanks for taking a look at the code.  This hopefully does the opposite.  I was seeing a crash in a debug message (an entity with no inventory was getting its inventory checked I think) and this change allows the message to be printed instead of crashing (the issue really wasn't about object types it was just trying to print out the object type for an entity with no inventory in a debug message resulting in a crash instead of a message).

Reply:
Why did this start happening all of a sudden?
You have to tackle issues like this at the root.

Reply:
This happen when I mess around testing spells in the starting room.  I assumed I was just triggering one of the many debug messages that you get while running (just this debug message was crashing TOEE).  I am not against eventually chasing down whatever is causing this but I doubt it is critical.  The unprotected message was however making things worse.  Now it should log the issue.

Reply:
The danger here is that in reality you're getting a corrupted object from somewhere, and it's obj_f_type is garbage because of the corruption. By just catching the conversion here, you'll never actually find the root cause for the corruption.

Reply:
I can look into this one a bit more and see if I can fix the underlying cause (or atleast add a bug report to fix later).  I should probably have some time for that on Tuesday.  I do still think this change is for the better though.  Log messages that can cause undefined behavior are very bad IMHO.  If the code went whack-a-do later, we would still have the log message to figure it out.

Reply:
Looks like ObjRegistry::Get() is returning a bad handle.  If I change the method back to using the "old way", it returns a null pointer instead of a bad pointer.  The crash happens when loading a saved game after the game has been started and one of the new spells is cast.  It seems that the particle system for the new spell effect is causing the query.  Do you know how the removal of the particle system happens for spells?

Reply:
Do you know how TOEE removes all the spell effects on shutdown?  Seems that an effect is getting removed too late in the shutdown process.  I am having some trouble tracking down how they actually get removed.

Reply:
I now have the root issue fixed in my latest push.